controller/users.go

```diff
@@ -77,6 +87,227 @@ func (c *UsersController) Show(ctx *app.ShowUsersContext) error {
 	})
 }
 
+// Create creates a user when requested using a service account token
+func (c *UsersController) Create(ctx *app.CreateUsersContext) error {
+
+	isSvcAccount := token.IsSpecificServiceAccount(ctx, []string{"online-registration"})
+	if !isSvcAccount {
+		log.Error(ctx, nil, "The account is not an authorized service account allowed to create a new user")
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("account not authorized to create users."))
+	}
+
+	tokenEndpoint, err := c.config.GetKeycloakEndpointToken(ctx.RequestData)
+	if err != nil {
+		return errors.NewInternalError(ctx, err)
+	}
+	log.Info(nil, map[string]interface{}{
+		"keycloak_client_id": c.config.GetKeycloakClientID(),
+		"token_endpoint":     tokenEndpoint,
+	}, "will generate PAT ")
+	protectedAccessToken, err := auth.GetProtectedAPIToken(ctx, tokenEndpoint, c.config.GetKeycloakClientID(), c.config.GetKeycloakSecret())
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+	}
+
+	keycloakUserID, err := c.createUserInKeycloak(ctx, protectedAccessToken)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to create user in keycloak")
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+	}
+	log.Info(ctx, map[string]interface{}{
+		"keycloak_user_id": keycloakUserID,
+	}, "successfully created new user in keycloak")
+
+	identityID, err := uuid.FromString(*keycloakUserID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+	}
+
+	identity, user, err := c.createUserInDB(ctx, identityID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to create user in DB")
+
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+	}
+
+	err = c.linkUserToRHD(ctx, *identity, protectedAccessToken)
```

Comment:
was ensuring we first sync kc <--> auth-db first before anything else :)

Reply:
If we do all the KC part first then Auth/WIT part then if anything bad happens in KC we will have a broken stuff in KC only (easy to clean up manually). Otherwise we may have broken stuff everywhere.